### DISCUSSION TITLE
The behavior of ScriptRuntime.ImportModule

### DISCUSSION THREAD

**Discussion by otogawakatsutoshi** [IronLanguages/ironpython3#1645] [Q&A] [answered]
When I try to import youtube-dl with ScriptRuntime.ImportModule using the ironPython3 librar at nuget.org, I get the error "IronPython.Runtime.Exceptions.ImportException : ImportException : No module named 'codecs'" error.

However, when I import youtube-dl using ironpython3 with interactive expression, no error occurs.

Is there a difference in behavior between ScriptRuntime.ImportModule and interactive import?

The assumption is that I have removed the line where "from \_\_future\_\_" is used to make youtube-dl work with ironpython3.

Translated with www.DeepL.com/Translator (free version)

**Comment by slozier** [comment, marked as answer]
Are you missing paths to the standard library? For example:
```cs
var engine = Python.CreateEngine();
var paths = engine.GetSearchPaths();
paths.Add(@"C:\Program Files\IronPython 3.4\Lib");
paths.Add(@"C:\Program Files\IronPython 3.4\Lib\site-packages"); // change this to be where you installed youtube-dl
engine.SetSearchPaths(paths);
engine.ImportModule("youtube_dl");
```

**Reply by otogawakatsutoshi** [reply]
Script worked! 
Thank you for teaching me how to do this!
